src/payment/payment-action-creator.spec.ts

```diff
@@ -249,5 +251,43 @@ describe('PaymentActionCreator', () => {
                 },
             ]);
         });
+
+        it('passes "set_as_default_stored_instrument" flag as null to the paymentRequestTransformer when vaulting, but not as default', () => {
+            const payment = getPayment();
+
+            paymentActionCreator.initializeOffsitePayment(payment.methodId, payment.gatewayId, undefined, true)(store);
+
+            expect(paymentRequestTransformer.transform).toHaveBeenCalledWith(expect.anything(), expect.anything());
+
+            expect(paymentRequestTransformer.transform).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    paymentData: expect.objectContaining({
+                        formattedPayload: expect.objectContaining({
+                            set_as_default_stored_instrument: null,
+                        }),
+                    }),
+                }),
+                expect.anything()
+            );
+        });
+
+        it('passes "set_as_default_stored_instrument" flag as true to the paymentRequestTransformer when vaulting and making default', () => {
+            const payment = getPayment();
+
+            paymentActionCreator.initializeOffsitePayment(payment.methodId, payment.gatewayId, undefined, true, undefined, undefined, true)(store);
+
+            expect(paymentRequestTransformer.transform).toHaveBeenCalledWith(expect.anything(), expect.anything());
```

Comment:
This line looks redundant to me with the next `expect().toHaveBeenCalledWith()` statement at line 281. Do we have it for a reason?

Reply:
Done